tests/nosetests/pyanaconda_tests/module_source_base_test.py

```diff
@@ -0,0 +1,122 @@
+#
+# Copyright (C) 2020  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import unittest
+from unittest.mock import patch
+
+from pyanaconda.core.constants import INSTALL_TREE
+from pyanaconda.modules.common.errors.payload import SourceSetupError
+from pyanaconda.modules.payloads.constants import SourceType
+from pyanaconda.modules.payloads.source.mount_tasks import SetUpMountTask, TearDownMountTask
+from pyanaconda.modules.payloads.source.source_base import MountingSourceBase
+
+mount_location = "/some/dir"
+
+
+class DummyMountingSourceSubclass(MountingSourceBase):
+    """Dummy class to test code in its abstract ancestor."""
+
+    @property
+    def type(self):
+        return SourceType.URL
+
+    def for_publication(self):
+        return None
+
+    def set_up_with_tasks(self):
+        return []
+
+
+class DummySetUpMountTaskSubclass(SetUpMountTask):
```

Comment:
In general, I don't think it is necessary to write tests with dummy implementations. It is enough the test one of the real implementations.